internal/app/singularity/registry_login.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2019-2023, Sylabs Inc. All rights reserved.
+// Copyright (c) 2020, Control Command Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package singularity
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/sylabs/singularity/internal/pkg/remote"
+	"github.com/sylabs/singularity/pkg/sylog"
+)
+
+// RegistryLogin logs in to an OCI/Docker registry.
+func RegistryLogin(usrConfigFile string, args *LoginArgs) (err error) {
+	// opening config file
+	file, err := os.OpenFile(usrConfigFile, os.O_RDWR|os.O_CREATE, 0o600)
+	if err != nil {
+		return fmt.Errorf("while opening remote config file: %s", err)
+	}
+	defer file.Close()
+
+	// read file contents to config struct
+	c, err := remote.ReadFrom(file)
+	if err != nil {
+		return fmt.Errorf("while parsing remote config data: %s", err)
+	}
+
+	if err := syncSysConfig(c); err != nil {
+		return err
+	}
+
+	if err := c.Login(args.Name, args.Username, args.Password, args.Insecure); err != nil {
+		return fmt.Errorf("while login to %s: %s", args.Name, err)
+	}
+
+	// truncating file before writing new contents and syncing to commit file
+	if err := file.Truncate(0); err != nil {
+		return fmt.Errorf("while truncating remote config file: %s", err)
+	}
+
+	if n, err := file.Seek(0, io.SeekStart); err != nil || n != 0 {
+		return fmt.Errorf("failed to reset %s cursor: %s", file.Name(), err)
+	}
+
+	if _, err := c.WriteTo(file); err != nil {
+		return fmt.Errorf("while writing remote config to file: %s", err)
+	}
+
+	if err := file.Sync(); err != nil {
+		return fmt.Errorf("failed to flush remote config file %s: %s", file.Name(), err)
+	}
+
+	sylog.Infof("Token stored in %s", file.Name())
```

Comment:
Nit - the token doesn't actually get stored in `remote.yaml` here. It goes into `~/.singularity/docker-config.json`